apps/hexagon_benchmarks/conv3x3.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+class Conv3x3 : public Generator<Conv3x3> {
+public:
+    // Takes an 8 bit image; one channel.
+    Input<Buffer<uint8_t>> input{"input", 2};
+    Input<Buffer<int8_t>> mask{"mask", 2};
+    // Outputs an 8 bit image; one channel.
+    Output<Buffer<uint8_t>> output{"output", 2};
+    Var x{"x"}, y{"y"};
+    Func bounded_input{"input_bounded"};
+    GeneratorParam<Type> accumulator_type{"accumulator_type", Int(16)};
```

Comment:
Would be nice to reorder and group these things. I think we usually do GeneratorParams first, params (inputs/outputs) second, and then put the Vars and intermediate funcs later (possibly even make them private? @steven-johnson what do you think would be good here?)